### ISSUE TITLE
ens3: bad gso: type: 1, size: 1432

### ISSUE THREAD

**Issue by berendt** [gardenlinux/gardenlinux#2278]
After upgrading to Gardenlinux 1592.0 we see a lot of those error message in dmesg. Network performance is degraded.

```
[  140.932766] ens3: bad gso: type: 1, size: 1432
[  141.036511] ens3: bad gso: type: 1, size: 1432
[  141.079216] ens3: bad gso: type: 1, size: 1432
```

Probably this is related to https://bugzilla.kernel.org/show_bug.cgi?id=219129.

We run on top of OpenStack (2023.2) with virtio enabled.

**Comment by fwilhe** [comment]
Hi @berendt 

thanks for reporting. If you are right that this is the mentioned issue, this should be fixed in linux 6.6.46. I don't know if that's actually the case, but this kernel should be part of our next patch release. I can't tell you when that will be available.

**Comment by berendt** [comment]
> Hi @berendt
> 
> thanks for reporting. If you are right that this is the mentioned issue, this should be fixed in linux 6.6.46. I don't know if that's actually the case, but this kernel should be part of our next patch release. I can't tell you when that will be available.

I don't know your policy here. But from my point of view, a very timely patch release would make sense here as probably all deployments involving OpenStack clusters for shoots are actually affected. It would be nice to be able to solve the problem promptly without having to go back to 1443.10.

**Comment by fwilhe** [comment]
Yes, I understand. ping @pnpavlov

**Comment by fwilhe** [comment]
Have you tested/investigated if 6.6.46 does mitigate the issue? 

We have binary packages available here https://github.com/gardenlinux/package-linux/releases/tag/6.6.46-0gardenlinux0, but no disk images (yet).

**Comment by pnpavlov** [comment]
hey @berendt , nice to "see" you again, sorry for the background for the "meeting"! 

Yes, we will update the kernel with #2273 . We are collecting the changes required and will work on patch next week.

**Comment by fwilhe** [comment]
Hi @berendt,

We won't be able to make a patch release today, but since the new kernel version is already in our nightly, and since the nightly version (1599 today) is still quite close to the release (1592), it would be super helpful for us if you could verify that the new kernel fixes the issue.

Here is the link to the github artefact for [openstack-gardener_prod-amd64-1599.0-db367054](https://github.com/gardenlinux/gardenlinux/actions/runs/10415427320/artifacts/1818847980).

**Comment by Vincinator** [comment]
The artefacts are also uploaded to S3, these should be publicly accessible: 
- s3://gardenlinux-github-releases/objects/openstack-gardener_prod-amd64-1599.0-db367054/
- s3://gardenlinux-github-releases/objects/openstackbaremetal-gardener_prod-amd64-1599.0-db367054/
- s3://gardenlinux-github-releases/objects/openstackbaremetal-gardener_prod-arm64-1599.0-db367054/

**Comment by pnpavlov** [comment]
@berendt , texted you in slack k8s space, can you please double check if the nightly run that @Vincinator  shared above does solve the problem and also if the arm64 image is good? if so, we will go forward with those two changes in the patch1 next week.

**Comment by berendt** [comment]
We are currently testing. It looks as if the messages in the kernel log no longer occur with the test image.

**Comment by fwilhe** [comment]
@berendt Thanks for checking 👍

**Comment by berendt** [comment]
We are still running a few tests. But the network performance now looks normal again.

**Comment by pnpavlov** [comment]
Thank you @berendt & team, we will go ahead and push that change in https://github.com/gardenlinux/gardenlinux/issues/2273. Should be published in next couple days together with couple other things.

**Comment by fwilhe** [comment]
Hi @berendt 

We released 1592.1 yesterday which includes kernel 6.6.47 with the fix you described.

Thanks for the heads up and please let us know if you find any other issue.

Best,
Florian
